packages/rules-unit-testing/src/api/index.ts

```diff
@@ -261,9 +286,23 @@ export function clearFirestoreData(
 
 export function assertFails(pr: Promise<any>): any {
   return pr.then(
-    v =>
-      Promise.reject(new Error('Expected request to fail, but it succeeded.')),
-    err => err
+    (v: any) => {
+      return Promise.reject(
+        new Error('Expected request to fail, but it succeeded.')
+      );
+    },
+    (err: any) => {
```

Comment:
We now make sure the error is a `FirebaseError` with a `message` that contains `PERMISSION_DENIED`.  This prevents connect failures from giving the illusion of security.